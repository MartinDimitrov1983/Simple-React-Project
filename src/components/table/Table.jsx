import React from 'react'
import TableRow from './TableRow.jsx'
import styles from './index.module.css'

const HEADER_VALUES = {
    id: 'id',
    name: 'name',
    plan: 'plan',
    units: 'units',
    city: 'city',
    region: 'region',
    manager: 'manager',
}
const Table = ({ data, onDelete, ...props }) => {
    return (
        <div className={styles.container} {...props}>
            <table>
                <thead>
                    <TableRow
                        tableCells={HEADER_VALUES}
                        tag="th"
                        isBodyRow={false}
                    />
                </thead>
                <tbody>
                    {data?.map((rowData) => (
                        <TableRow
                            key={rowData.id}
                            tableCells={rowData}
                            tag="td"
                            onDelete={onDelete}
                        />
                    ))}
                </tbody>
            </table>
        </div>
    )
}

export default Table
