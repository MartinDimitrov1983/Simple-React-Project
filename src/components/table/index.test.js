import React from 'react'
import { render } from '@testing-library/react'

import Table from './Table'

test('Snapshot Table', () => {
    const { asFragment } = render(<Table />)
    expect(asFragment()).toMatchSnapshot()
})
