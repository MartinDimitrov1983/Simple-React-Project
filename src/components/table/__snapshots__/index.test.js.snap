// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Table 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <table>
      <thead>
        <tr>
          <th>
            name
          </th>
          <th>
            plan
          </th>
          <th>
            units
          </th>
          <th>
            city
          </th>
          <th>
            region
          </th>
          <th>
            manager
          </th>
          <th />
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
